errbot/backends/slack.py

```diff
@@ -157,6 +157,15 @@ def aclattr(self):
         # an incorrect format from SlackMUCOccupant.
         return "@%s" % self.username
 
+    @property
+    def email(self):
+        """Convert a Slack user ID to their user email"""
+        user = self._sc.server.users.find(self._userid)
+        if user is None:
+            log.error("Cannot find user with ID %s" % self._userid)
+            return "<%s>" % self._userid
```

Comment:
this is a strange return, how the calling code knows it could not find an email?

Reply:
What do you think should be returned?
By the way, it's very unlikely for a slack user not to have an email :)

Reply:
@gbin any update on this?

Reply:
@gbin You still working on this?

Reply:
I suppose the real question is... does all user objects return an email address? Is there a possibility where a user is found but the `user.email` attribute is `None` or not valid?

Reply:
The bot will need a special scope to be able to access them if I understand this correctly.

>If you want access to email addresses, you'll need the new OAuth permission scope, users:read.email. It provides an explicit, additive way to request access to team email addresses.

>Additionally, the bot scope will no longer grant bot user tokens access to email addresses. Bot users must utilize a user token and the users:read.email scope instead.

https://api.slack.com/changelog/2017-04-narrowing-email-access

Reply:
I wonder if this is just for the Web API methods and how this affects RTM

Reply:
My question was a contract question...
If the user is not found, the property should be None and not "<id>".

Imagine you want to parse the email...

Reply:
@gbin Very good point, I would say merge this and we can address/clean-up the concern with no user found and trying to return an id with another PR as there are other methods that follow this return format.